roles/ceph-nfs/tasks/pre_requisite.yml

```diff
@@ -1,30 +1,33 @@
 ---
-- name: create nfs gateway directories
-  file:
-    path: "{{ item }}"
-    state: directory
-    owner: "ceph"
-    group: "ceph"
-    mode: "0755"
-  with_items:
-    - /var/lib/nfs/ganesha
-    - /var/run/ganesha
-
 - name: create rgw nfs user
   command: radosgw-admin --cluster {{ cluster }} user create --uid={{ ceph_nfs_rgw_user }} --display-name="RGW NFS User"
   register: rgwuser
+  delegate_to: "{{ groups[mon_group_name][0] }}"
   when: nfs_obj_gw
 
 - name: set access key
   set_fact:
     ceph_nfs_rgw_access_key: "{{ (rgwuser.stdout | from_json)['keys'][0]['access_key'] }}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
   when: nfs_obj_gw
 
 - name: set secret key
   set_fact:
     ceph_nfs_rgw_secret_key: "{{(rgwuser.stdout | from_json)['keys'][0]['secret_key']}}"
+  delegate_to: "{{ groups[mon_group_name][0] }}"
   when: nfs_obj_gw
 
+- name: create nfs gateway directories
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "ceph"
+    group: "ceph"
+    mode: "0755"
+  with_items:
+    - /var/lib/nfs/ganesha
+    - /var/run/ganesha
+
 - name: generate ganesha configuration file
   action: config_template
   args:
```

Comment:
Is there any reasons why the template is not part of the `ceph-nfs` role instead?